Added 'upgrades fever' server command.

arbuckle opened the pull request:
This command initiates an upgrade fever event. Design is intended for FFA but it works for CTF as well.

- new command 'server upgrades fever' starts the event
- current player upgrades are stored and values reset to 5
- on respawn, config.upgrades.fever is checked to max upgrades
- bots only get upgraded to 3/5
- event also ends with 'server upgrades fever'. it's a toggle

The only other thing I'm thinking of here is about how the fever event might be toggled from `server/periodic`. For FFA it could be fun to run an event every hour or so, for a few minutes. If that were the case I'd want to pull the logic out of server.js and into a separate function, and conditionally modify the welcome message so that new players are informed about the special condition.

wight-airmash commented:
> bots only get upgraded to 3/5

Heli bots with 3 defense needs [2 predator missiles](https://github.com/wight-airmash/ab-server/blob/master/docs/damage.md#predator-missile-hits) to be killed, I don't know how it would be in FFA, but in CTF the experiments felt unbalanced.

>  For FFA it could be fun to run an event every hour or so, for a few minutes.

There are `TIMELINE_CLOCK_HOUR` and `TIMELINE_CLOCK_MINUTE` events which are emitted every hour and every minute respectively. For example, listen to the "hour" event, begin incrementing counter every minute with the "minute" event till it reach desired value and then stop the fever.

> new players are informed about the special condition

You can listen to `PLAYERS_CREATED` event and inform just connected players like this:

```javascript
this.channel(CHANNEL_CHAT).delay(BROADCAST_CHAT_SERVER_WHISPER, playerId, message);
```

This code will whisper the connected player with safe time delay (starmash anti-spam filter).

arbuckle commented:
Thanks for the feedback.  I adjusted the implementation a little bit to reduce duplication and address a few defects.  I'll save periodic fevers for a different PR.  

- consolidated application of fever into a single function
- fixed issue where new connections didn't get upgraded
- server whispers to players that an event is ongoing
- bots nerfed to 2 defense upgrades
